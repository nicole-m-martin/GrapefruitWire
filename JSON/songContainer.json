{
    "lyrics": "Remember those walls I built\r\nWell baby they're tumbling down\r\nAnd they didn't even put up a fight\r\nThey didn't even make a sound\r\nI found a way to let you in\n\nBut I never really had a doubt\n\nStanding in the light of your halo\n\nI've got my angel now\n\n\n\nIt's like I've been awakened\n\nEvery rule I had you breakin'\n\nIt's the risk that I'm takin'\n\nI ain't never gonna shut you out\n\n\n\nEverywhere I'm looking now\n\nI'm surrounded by your embrace\n\nBaby I can see your halo\n\nYou know you're my saving grace\n\nYou're everything I need and more\n\nIt's written all over you face\n\nBaby I can feel your halo\n\nPray it won't fade away\n\n\n\n(I can feel your halo... halo...) halo\n\n(I can see your) halo... (halo...) halo-o\n\n(I can feel your) halo... (halo...) halo-o-o\n\n(I can see your) halo... (halo...) halo... uhoo\n\n\n\nHit me like a ray of sun\n\nBurning through my darkest nights\n\nYou're the only one that I want\n\nThink I'm addicted to your light\n\n\n\nI swore I'd never fall again\n\nBut this don't even feel like fallin'\n\nGravity can't forget\n\nTo pull me back to the ground again\n\n\n\nIt's like I've been awakened\n\nEvery rule I had you breakin'\n\nIt's the risk that I'm takin'\n\nI ain't never gonna shut you out\n\n\n\nEverywhere I'm looking now\n\nI'm surrounded by your embrace\n\nBaby I can see you halo\n\nYou know you're my saving grace\n\nYou're everything I need and more\n\nIt's written all over you face\n\nBaby I can feel your halo\n\nPray it won't fade away (#)\n\n\n\n(I can feel your halo... halo...) halo\n\n(I can see your) halo... (halo...) halo-o\n\n(I can feel your) halo... (halo...) halo-o-o\n\n(I can see your) halo... (halo...) halo\n\n\n\n(I can feel your halo [uhoo] halo...) halo\n\n(I can see your) halo... (halo...) halo-o\n\n(I can feel your) halo... (halo...) halo-o-o\n\n(I can see your) halo... (halo...) halo... uhoo\n\n\n\nHalo uhooo\n\nhalo uhooo uhooo uhoo-oo-oo-oo-oo-oo\n\nuhoo-oo-oo-oo-oo-oo-oo oo oo\n\n\n\nEverywhere I'm looking now\n\nI'm surrounded by your embrace\n\nBaby I can see you halo\n\nYou know you're my saving grace\n\nYou're everything I need and more\n\nIt's written all over you face\n\nBaby I can feel your halo\n\nPray it won't fade away\n\n\n\n(I can feel your halo... halo...) halo\n\n(I can see your) halo... (halo...) halo-o\n\n(I can feel your) halo... (halo...) halo-o-o\n\n(I can see your) halo... (halo...) halo\n\n\n\n(I can feel your halo [uhoo] halo...) halo\n\n(I can see your) halo... (halo...) halo-o\n\n(I can feel your) halo... (halo...) halo-o-o\n\n(I can see your) halo... (halo...) halo... uhoo..."
    }