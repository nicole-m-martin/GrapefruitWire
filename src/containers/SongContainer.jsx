import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router';
import SongLyricsItem from '../components/song/SongLyricsItem';
import { fetchLyrics } from '../services/lyrics-api';

const SongContainer = () => {
    const [loading, setLoading] = useState(true);
    const [lyrics, setLyrics] = useState('');

    const { artistName, title } = useParams();

    useEffect(() => {
        fetchLyrics(artistName, title)
            .then(setLyrics)
            .finally(() => setLoading(false));
    }, []);
    
    return loading ? (
        <h2>Loading...</h2>
    ) : (
        <main>
            <SongLyricsItem lyrics={lyrics}/>
        </main>
    )
}

export default SongContainer;
