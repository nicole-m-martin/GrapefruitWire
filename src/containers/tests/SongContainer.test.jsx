import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import songContainer from '../../../JSON/songContainer.json';
import SongContainer from '../SongContainer';

const server = setupServer(
  rest.get('https://api.lyrics.ovh/v1/beyonce/halo', (req, res, ctx) => {
    return res(ctx.json(songContainer));
  })
);

describe('song container', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('fetches song lyrics and renders them to the page', () => {
    render(
      <MemoryRouter>
        <SongContainer />
      </MemoryRouter>
    );

    const ul = screen.findByRole('div', { name: 'lyric' });

    expect(ul).toMatchSnapshot();
  });
});
