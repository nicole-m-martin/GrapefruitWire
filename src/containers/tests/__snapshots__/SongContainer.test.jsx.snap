// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`song container fetches song lyrics and renders them to the page 1`] = `Promise {}`;
